### PR TITLE
#4844 feat: renew id token when expires

### LINKED ISSUE #4844: Offer users to reconnect when id_token expires
The following error occurs when a JWT token has expired when requesting for `api/account/get`. 

I think one way to address this is by going to FlowCrypt settings -> resetting the account -> sign back in to refresh the id_token needed by the backend (not user friendly). 

The customer wonders if there's something we can do to simplify it. 

Reference: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGrbbxhKWXXkjFGnRLcMxlLkRQc

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR renews id token when expires

close #4844 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Difficult to test (Hard/impossible to reproduce in semaphoreci)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/api/shared/api.ts [hunk 1]
```diff
@@ -205,8 +206,23 @@ export class Api {
...
+    } catch (firstAttemptErr) {
+      const idToken = req.headers?.Authorization?.split(' ')[1];
+      if (ApiErr.isAuthErr(firstAttemptErr) && idToken) {
+        // force refresh token
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const { email } = GoogleAuth.parseIdToken(idToken);
+        if (email) {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          req.headers!.Authorization = await GoogleAuth.googleApiAuthHeader(email, true);
+          return await Api.ajax(req, Catch.stackTrace()) as RT;
```
Comment (ioanmo226) [inline extension/js/common/api/shared/api.ts:221]: As `Api` class uses `GoogleAuth` and `GoogleAuth` uses `Api` class, circular dependency error occurs.
So I'm thinking to move common method(parseIdToken, googleApiAuthHeader, googleAuthRefreshToken, googleAuthSaveTokens) to new common class(such as GoogleAuthExtension)

@sosnovsky Do you have better suggestion?

Comment (sosnovsky) [reply extension/js/common/api/shared/api.ts:221]: I think proposed solution should be good, we can call this new class `GoogleAuthHelper`

Comment (ioanmo226) [reply extension/js/common/api/shared/api.ts:221]: Thanks

File: extension/js/common/api/email-provider/gmail/google-auth-helper.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        crossDomain: true,
+        async: true,
+      },
+      Catch.stackTrace()
+    )) as unknown as GoogleAuthTokensResponse;
+  };
+
+  /// Duplicated code begin
+  // This logic was duplicated (It's exactly same as Api.ajax) but added here because we can't use Api.ajax here as it will cause circular dependency warning
+  private static ajax = async (req: JQueryAjaxSettings, stack: string): Promise<unknown | JQuery.jqXHR<unknown>> => {
```
Comment (ioanmo226) [inline extension/js/common/api/email-provider/gmail/google-auth-helper.ts:174]: Let me know your thought about this block if you agree on duplicating logi or not.

Comment (ioanmo226) [reply extension/js/common/api/email-provider/gmail/google-auth-helper.ts]: Sounds good.

**Pull request closed without merging**